tests/mr/device/aligned_mr_tests.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <rmm/detail/error.hpp>
+#include <rmm/device_buffer.hpp>
+#include <rmm/mr/device/aligned_resource_adaptor.hpp>
+#include <rmm/mr/device/tracking_resource_adaptor.hpp>
+
+#include <gtest/gtest.h>
+
+namespace rmm::test {
+namespace {
+
+using tracking_adaptor = rmm::mr::tracking_resource_adaptor<rmm::mr::device_memory_resource>;
+using aligned_adaptor  = rmm::mr::aligned_resource_adaptor<tracking_adaptor>;
+
+TEST(AlignedTest, ThrowOnNullUpstream)
+{
+  auto construct_nullptr = []() { aligned_adaptor mr{nullptr}; };
+  EXPECT_THROW(construct_nullptr(), rmm::logic_error);
+}
+
+TEST(AlignedTest, SmallAllocations)
+{
+  tracking_adaptor tracker{rmm::mr::get_current_device_resource()};
+  aligned_adaptor mr{&tracker};
```

Comment:
I would expect adaptors made with really high and low alignment requirements and verification that the returned pointers were aligned properly. Take care to exercise the edge cases by allocating `alignment + small_val` 
and `alignment - small_val` bytes.

Reply:
Done.